src/main/java/org/springframework/integration/splunk/support/SplunkHECWriter.java

```diff
@@ -0,0 +1,459 @@
+					ioReactor, sslSessionStrategy);
+			cm.setMaxTotal(getPoolsize());
+
+			HttpHost splunk = new HttpHost(getHost(), getPort());
+			cm.setMaxPerRoute(new HttpRoute(splunk), getPoolsize());
+
+			httpClient = HttpAsyncClients.custom().setConnectionManager(cm)
+					.build();
+
+			uri = new URIBuilder().setScheme(isHttps() ? "https" : "http")
+					.setHost(getHost()).setPort(getPort())
+					.setPath("/services/collector").build();
+
+			httpClient.start();
+
+			if (isBatchMode()) {
+				new BatchBufferActivityCheckerThread(this).start();
+			}
+		} catch (Exception e) {
+		}
+
+		this.running = true;
+
+	}
+
+	@Override
+	public void stop() {
+		if (!running) {
+			return;
+		}
+
+		try {
+			httpClient.close();
+
+		} catch (Exception e) {
+		}
+
+		this.running = false;
+
+	}
+
+	private String wrapMessageInQuotes(String message) {
+
+		return "\"" + message + "\"";
+	}
+
+	class BatchBufferActivityCheckerThread extends Thread {
+
+		SplunkHECWriter parent;
+
+		BatchBufferActivityCheckerThread(SplunkHECWriter parent) {
+
+			this.parent = parent;
+		}
+
+		public void run() {
+
+			while (true) {
+				String currentMessage = "";
+				try {
+					long currentTime = System.currentTimeMillis();
+					if ((currentTime - lastEventReceivedTime) >= getMaxInactiveTimeBeforeBatchFlush()) {
+						if (batchBuffer.size() > 0) {
+							currentMessage = rollOutBatchBuffer();
+							batchBuffer.clear();
+							currentBatchSizeBytes = 0;
+							hecPost(currentMessage);
+						}
+					}
+
+					Thread.sleep(1000);
+				} catch (Exception e) {
+					try {
+						parent.stop();
+					} catch (Exception e1) {
+					}
+
+					try {
+						parent.start();
+					} catch (Exception e2) {
+					}
+				}
+
+			}
+		}
+	}
+
+	private SSLContext getSSLContext() {
+		TrustStrategy acceptingTrustStrategy = new TrustStrategy() {
+			public boolean isTrusted(X509Certificate[] certificate,
+					String authType) {
+				return true;
+			}
+		};
+		SSLContext sslContext = null;
+		try {
+			sslContext = SSLContexts.custom()
+					.loadTrustMaterial(null, acceptingTrustStrategy).build();
+		} catch (Exception e) {
+			// Handle error
+		}
+		return sslContext;
+
+	}
+
+	private boolean flushBuffer() {
+
+		return (currentBatchSizeBytes >= getMaxBatchSizeBytes())
+				|| (batchBuffer.size() >= getMaxBatchSizeEvents());
+
+	}
+
+	private String rollOutBatchBuffer() {
+
+		StringBuffer sb = new StringBuffer();
+
+		for (String event : batchBuffer) {
+			sb.append(event);
+		}
+
+		return sb.toString();
+	}
+
+	private void hecPost(String currentMessage) throws Exception {
+
+		HttpPost post = new HttpPost(uri);
+		post.addHeader("Authorization", "Splunk " + getToken());
+
+		StringEntity requestEntity = new StringEntity(currentMessage,
+				ContentType.create("application/json", "UTF-8"));
+
+		post.setEntity(requestEntity);
+		Future<HttpResponse> future = httpClient.execute(post, null);
```

Comment:
I'm pretty sure we can just rely on the `RestTemplate` abstraction and don't bring such a low-level API into our one.
Everything else can be configured outside on the `RestTemplate` bean.